ctgan/data_transformer.py

```diff
@@ -36,43 +33,53 @@ def __init__(self, max_clusters=10, weight_threshold=0.005):
         self._max_clusters = max_clusters
         self._weight_threshold = weight_threshold
 
-    def _fit_continuous(self, column_name, raw_column_data):
-        """Train Bayesian GMM for continuous column."""
-        gm = BayesianGaussianMixture(
-            n_components=self._max_clusters,
-            weight_concentration_prior_type='dirichlet_process',
-            weight_concentration_prior=0.001,
-            n_init=1
-        )
+    def _fit_continuous(self, data):
+        """Train Bayesian GMM for continuous columns.
 
-        gm.fit(raw_column_data.reshape(-1, 1))
-        valid_component_indicator = gm.weights_ > self._weight_threshold
-        num_components = valid_component_indicator.sum()
+        Args:
+            data (pd.DataFrame):
+                A dataframe containing a column.
+
+        Returns:
+            namedtuple:
+                A ``ColumnTransformInfo`` object.
+        """
+        gm = BayesGMMTransformer()
+        gm.fit(data, list(data.columns))
+        num_components = sum(gm.valid_component_indicator)
 
         return ColumnTransformInfo(
-            column_name=column_name, column_type='continuous', transform=gm,
-            transform_aux=valid_component_indicator,
+            column_name=data.columns[0], column_type='continuous', transform=gm,
             output_info=[SpanInfo(1, 'tanh'), SpanInfo(num_components, 'softmax')],
             output_dimensions=1 + num_components)
 
-    def _fit_discrete(self, column_name, raw_column_data):
-        """Fit one hot encoder for discrete column."""
-        ohe = OneHotEncodingTransformer()
-        fit_data = pd.DataFrame(raw_column_data, columns=[column_name])
+    def _fit_discrete(self, data):
+        """Fit one hot encoder for discrete column.
+
+        Args:
+            data (pd.DataFrame):
+                A dataframe containing a column.
 
-        ohe.fit(fit_data, column_name)
+        Returns:
+            namedtuple:
+                A ``ColumnTransformInfo`` object.
+        """
+        ohe = OneHotEncodingTransformer()
+        ohe.fit(data, list(data.columns))
         num_categories = len(ohe.dummies)
 
         return ColumnTransformInfo(
-            column_name=column_name, column_type='discrete', transform=ohe,
-            transform_aux=None,
+            column_name=data.columns[0], column_type='discrete', transform=ohe,
```

Comment:
minor: we could capture `column_name = data.columns[0]` at the top and use that here and on line 68. I think it just helps with readability. The same thing could be done in `_fit_continuous`